### ISSUE TITLE
Which INSPIRE theme covers power stations ?

### ISSUE THREAD

**Issue by fantamir** [INSPIRE-MIF/helpdesk#49]
To which INSPIRE theme the dataset representing power stations belongs? I supposed, the theme III-8-PF Production and Industrial Facilities is the right one for this purpose, but, on the other hand, theme III-20-ER Energy resources is used in certain cases as well.
Compare for example  
https://inspire-geoportal.ec.europa.eu/results.html?country=de&view=details&theme=pf 
and 
https://inspire-geoportal.ec.europa.eu/results.html?country=de&view=details&theme=er
Windkraftanlagen and Biogasanlagen are put by one country (Germany) to different INSPIRE themes, depending on the country region.
Maybe, the option could be also to categorize the renewable sources power stations to the ER theme and fossil + nuclear fuel  power stations to the PF theme, but I am not sure if such access is correct... 

Thank you very much
Miroslav Fanta
CENIA, Czech Environmental Information Agency

**Comment by JohannaOtt** [comment]
Hi Miroslav,

First of all I agree that this is not an easy decision and would also appreciate feedback from the respective expert groups. I am specifically stuggling with wind turbines. The source data contains information such as coordinates, legal status, height, rated power.
Here are my thoughts based on the documentation available:

**Energy Resources**
pro:
* RenewableAndWasteResource.Capacity is defined as "Energy capacity of a renewable energy resource within the spatial object.", description "NOTE the unit of measure to be used is MWe for electrical generation capacity and MWt for heat generation capacity." 
* The data specification contains this comment in 5.5.1.2: "Whereas renewable resources can only be quantified by expressing the capacity of the facility extracting the energy resource, the present amount of fossil fuel is subject to a domain-specific approach.". These two sounds as facilities for energy extraction (=wind turbines) are expected here
contra:
* The definition of VectorEnergyResource (from which ReneableAndWasteResource inherits) is "A vector spatial object defining an inferred or observable spatial extent of a resource that can be or has been used as a source of energy". This sounds as only the resource itself (=wind conditions in a specific area) is expected and not the facility extracting energy.
* The code "[wind](https://inspire.ec.europa.eu/codelist/RenewableAndWasteValue/wind)" that can be used for RenewableAndWasteResource.typeOfResource is defined as "Kinetic energy of wind exploited for electricity generation in wind turbines.". But the rated power of wind turbines describes the electric energy that can be exploted not the kinetic energy that is exploited - this again sounds more like the wind conditions are described here.

**Production and Industrial Facilites** 
pro:
* Definition of ProductionFacility is "One or more installations on the same site operated by the same natural or legal person, designed, built or installed to serve specific production or industrial purposes, comprehending all infrastructure, equipment and materials." - covers wind turbines from my understanding
* Data specification 2.2 states: "The overall set of activities considered within the Production and Industrial Facilities theme spans from extraction of resources, to their transformation in products or by-products, and their storage. [...] Transformation of resources should be seen both as transformation of one resource or product into another, or as transformation into energy, thus including power generation plants within the scope of this theme." -> considering wind turbines being power plants, this would mean that they belong to PF
* The codelist used for ProductionFacility.function.Function.activity contains the value "[Production of electricity](https://inspire.ec.europa.eu/codelist/EconomicActivityNACEValue/D.35.11)" which would be a fit for wind turbines
contra:
* The codelist used for ProductionFacility.function.Function.input and output only contains a matching value for the output "[Electricity](https://ec.europa.eu/eurostat/ramon/nomenclatures/index.cfm?TargetUrl=DSP_NOM_DTL_VIEW&StrNom=CPA_2008&StrLanguageCode=EN&IntPcKey=20997638&IntKey=20997741&StrLayoutCode=HIERARCHIC&IntCurrentPage=1)" but none for the input - this is not an issue as the model allows to not deliver an input but it makes me doubt if wind turbines are expected here.


It would be great if someone from the wind turbine community and/or the thematic working groups of the two themes could provide feedback to this issue.


All the best

Johanna

**Comment by heidivanparys** [comment]
Would the utility networks themes be an option? A utility network element has an attribute utility facility reference, referring to an [activity complex](https://inspire.ec.europa.eu/featureconcept/ActivityComplex), see also https://inspire.ec.europa.eu/data-model/approved/r4618-ir/html/index.htm?goto=2:3:20:3:1:8885.

Tagging one of the authors of the Dutch data specifications regarding utilities (IMKL), based on INSPIRE, perhaps @PalmJanssen can help?

**Comment by JohannaOtt** [comment]
@heidivanparys I would not think that utility networks is an option here because of what is written in chapter 2.2.1 of the data specification:

> Utility services and networks include the physical constructions for transport of defined utility products (namely pipelines for transport of oil, gas, chemicals, water, sewage and thermal products), transmission lines and cables (included those for transmission of electricity, phone and cable-TV signals) and other network elements for encasing pipes and cases (e.g. ducts, poles and towers).
> All kinds of transmission utility systems have nodes (e.g. pump stations), and they are linked to facilities for production and treatment of different kinds of utility products. These major production and treatment sites are described in the theme production and industrial facilities (Annex.III – PF).
> Six important types of utility networks are distinguished, namely Electricity Network, Oil, Gas & Chemicals Network, Sewer Network, Telecommunications Network, Thermal Network and Water Network.
> All these networks use the node-arc-node model, as defined in the Generic Network Model. Especially designed for a structured modelization of utility networks that allows its business use (estimation of propagation, calculation of capacities, etc.), the use of this node-arc-node model is also adapted for non-topological data, since the relation between nodes and arcs is optional.

As the datasets we are talking about here only contain wind turbines or similar power plants, I would not expect them in a schema where all networks should use the node-arc-node model.

**Comment by PalmJanssen** [comment]
@heidivanparys
@JohannaOtt 

About the use of activity complex:
- We use the the utility networks model for providing data in the use case for preventing excavation damage to the utility network
- The networks model includes a reference to the Activity Complex for referencing network elements to large complex network components like powerstations, water cleaning complexes and the like.
- We don't use this reference in our dedicated model because this info is not needed in our use case
- The ActivityComplex is not a feature type in the Utility networks model, but a general basemodel for describing a single economical, organisational unit. Which of course a powerplant is, but not viewed from its technical, material and usage perspective.
- So I would not use activity complex for representing a powerstation if more dedicated feature types are available
Paul

**Comment by heidivanparys** [comment]
@PalmJanssen Thank you for your thoughts on this!

> I am specifically stuggling with wind turbines.

@JohannaOtt My colleague @laers reminded me that that wind turbines can be provided in the [Buildings theme](https://inspire.ec.europa.eu/data-model/approved/r4618-ir/html/index.htm?goto=2:3:2:2:7911), the code list for [building nature](https://inspire.ec.europa.eu/codelist/BuildingNatureValue) has [wind turbine](https://inspire.ec.europa.eu/codelist/BuildingNatureValue/windTurbine) as one of its values.

**Comment by JohannaOtt** [comment]
@PalmJanssen Thanks for the input - in my opinion there is no more dedicated feature type available than ProductionFacility (which is an ActivityComplex).

@heidivanparys I also considered Buildings but as the dataset is about wind turbines as power plants and not as specific buildings - we should take into account that the data specification for buildings states in chapter 5.3.1.1.4: "the attribute buildingNature that focuses on the physical aspect of the building; however, this physical aspect is often expressed as a function (e.g. stadium, silo, windmill); this attribute aims to fulfil mainly mapping purposes and addresses only specific, noticeable buildings. This is a rather short and simple list of possible values, **with focus on two international use cases: air flights where buildings may be obstacles and marine navigation where buildings may be landmarks.**"
Also, the same question occurs for solar power plants, biogas etc that do not have specific buildingNature values. I think it makes more sense if the community agrees to a handling that leads to all (at least renewable) power plants ending up in the same theme/feature type.
This is why I do not think that Building is a good fit here.

**Comment by sMorrone** [comment]
Dear all,
I would say that there is no" one size fits all", here. The most suitable INSPIRE theme depends on the source data to transform, I mean on the specific attributes that characterise the source data sets.  So, if one needs to harmonise a datasets that focuses on the buildings as such, than the BU theme would be fine, conversely BU theme is not suitable if one needs to characterise the facility with the type of activity, the input and the output (then PF theme is more suitable).

**Comment by KathiSchleidt** [comment]
Just a short note supporting @JohannaOtt statement on 
`I think it makes more sense if the community agrees to a handling that leads to all (at least renewable) power plants ending up in the same theme/feature type.`

While I also agree with @sMorrone that there's no `one size fits all` here, I also (maybe naively) believe that similar topics across MS could/should have similar fits (e.g. similar available data). I DO see utilization of the same model across MS essential if we're aiming for supporting cross border use cases!

I'm at a loss how to organize such an approach, maybe direct targeting of known data providers searching for commonalities and ideally agreement?

**Comment by sMorrone** [comment]
@KathiSchleidt you are right `utilization of the same model across MS essential if we're aiming for supporting cross border use cases!` . Theoretically similar source data should lead to selection of same target data theme, but, unfortunately, theory is so different from practice. 
However, even the proposed approach` direct targeting of known data providers searching for commonalities and ideally agreement ` is really difficult to put into practice, though I see it feasible in particular cases.
For example, I see interesting the approach used for the Priority datasets (and hope such an approach could be used for the HVD list, when this will be available): to have an indication (in the case of PDS this comes from the EEA) of the mapping to be adopted per each addressed source dataset.

**Comment by KathiSchleidt** [comment]
So one would need a parallel stakeholder to EEA but for the energy community - any insights who this could be from those providing this data?

**Comment by JohannaOtt** [comment]
> I would say that there is no" one size fits all", here. The most suitable INSPIRE theme depends on the source data to transform, I mean on the specific attributes that characterise the source data sets.

> While I also agree with @sMorrone that there's no one size fits all here, 

@sMorrone @KathiSchleidt Agreed. I was referring to similar dataset representing/focusing on wind turbines as power plants in my comment - should have made that point clearer I guess.


> conversely BU theme is not suitable if one needs to characterise the facility with the type of activity, the input and the output (then PF theme is more suitable)

@sMorrone This is also my assessment but there is for example no attribute to represent the capacity or height of the wind turbines which is typical information and gets lost with this modeling. This is why I was asking if any of the contributors of the different data specifications has any more input on if power plants were considered during model creation.

**Comment by fantamir** [comment]
Dear colleagues,

thank you very much for your contribution to the discussion. To be honest, I am surprised, that the conclusion is not unequivocal.

With regard to the standpoint of the experts from the Czech Office for Surveying, Mapping and Cadastre, we have decided to use the INSPIRE theme **III-8-PF Production and Industrial Facilities** for power stations dataset. One of the key arguments is also the occurence of the mandatory attribute "**function**" in the ProductionFacility type of the III-8-PF theme data model. According to the documentation available, the **NACE code** is preferred in this attribute for the description of the activity. Next, the NACE codelist contains an item D.35.11 "**Production of electricity**" (see [https://inspire.ec.europa.eu/codelist/EconomicActivityNACEValue/D.35.11]). 
The wording of its definition is following: "_This class includes: operation of generation facilities that produce electric energy; including thermal, nuclear, hydroelectric, gas turbine, diesel and **renewable**. This class excludes: production of electricity through incineration of waste, see 38.21_".

I fully agree with the comment, that specific attributes important for the potential users of the power stations dataset are missing in the theme III-8-PF data model (e.g. type of the energy source (gas, wind, water, ...), power station performance, height in the case of wind turbines, ...). Since the proposal and implementation of the INSPIRE data model extension seems to be a complicated issue, we intend to solve this problem by simple adding of the additional parameters to the data model (modification of the XSD schema downloaded form the INSPIRE web pages). We have already tested this access in two other INSPIRE themes and the resulting GML files were succesfully tested by the INSPIRE validator.
I am understanding, that such extension of the data model is not "official", but the process is fast and easy, list of added attributes can be quicky modified with regard to the content of original source data and harmonised output GML data are accepted by validator. The newly added parameters and their content have to be easily understable for potential data users, of course. To summarize this description - data are harmonised with INSPIRE requirements and, in addition, no important attributes are lost and the users will immediately understand the extended structure of the data.

**Comment by KathiSchleidt** [comment]
@fantamir to my understanding extending INSPIRE models by adding attributes is not THAT difficult (seems that on a technical level you can handle, are only worried on the administrative aspects of this work). Be careful NOT to confuse extending INSPIRE Schemas with the requirements to a Good Practice (different issue!)

You mention that you'd extended INSPIRE XSDs had successfully validated these - could you provide more info on this? I've long been curious how the Validator deals with such extensions!

**Comment by fantamir** [comment]
Dear Kathi,

attached you can find the XSD schema for INSPIRE theme III/7 Environmental Monitoring Facilities with 4 additional attributes (file EnvironmentalMonitoringFacilities_ExtendedCZ_test.xsd, rows 553 - 567) and a sample of harmonised GML file based on this XSD schema (file r18AQ_pollutants_sample1_GML_test.gml, additional attributes are on rows 52-55 and 80-83).

Validation result is available here: https://inspire.ec.europa.eu/validator/test-run/details.html?id=EID0f47ff42-8616-49d8-8025-de91e6032afc. Without any connection with the data model modification, the test report indicates coordinate reference system error - see https://github.com/INSPIRE-MIF/helpdesk-validator/issues/617 for details.

[III-7-EF_DataModel_AdditionalAttributes.zip](https://github.com/INSPIRE-MIF/helpdesk/files/7259350/III-7-EF_DataModel_AdditionalAttributes.zip)

**Comment by KathiSchleidt** [comment]
Hi Miroslav,

interesting approach! I'd have provided these yearly values as measurements from the EF, but having it directly integrated could be useful for various use cases.

However, I worry a bit about your schema modification - validates at present as Annex III Themes are not yet being checked against the Schema, but do worry about what will happen once this is integrated to the validator.

I've done a quick hack, extended EF to EF_CZ, that provided the additional attributes you've proposed:
[EF_CZ.zip](https://github.com/INSPIRE-MIF/helpdesk/files/7259776/EF_CZ.zip)

:?
